crates/ruff_linter/src/rules/flake8_use_pathlib/rules/os_chmod.rs

```diff
@@ -73,22 +78,85 @@ pub(crate) fn os_chmod(checker: &Checker, call: &ExprCall, segments: &[&str]) {
     //           0     1          2               3
     // os.chmod(path, mode, *, dir_fd=None, follow_symlinks=True)
     // ```
-    if call
-        .arguments
-        .find_argument_value("path", 0)
-        .is_some_and(|expr| is_file_descriptor(expr, checker.semantic()))
+    let path_arg = call.arguments.find_argument_value("path", 0);
+
+    if path_arg.is_some_and(|expr| is_file_descriptor(expr, checker.semantic()))
         || is_keyword_only_argument_non_default(&call.arguments, "dir_fd")
     {
         return;
     }
 
-    check_os_pathlib_two_arg_calls(
-        checker,
-        call,
-        "chmod",
-        "path",
-        "mode",
-        is_fix_os_chmod_enabled(checker.settings()),
-        OsChmod,
-    );
+    let range = call.range();
+    let mut diagnostic = checker.report_diagnostic(OsChmod, call.func.range());
+
+    if !is_fix_os_chmod_enabled(checker.settings()) {
+        return;
+    }
+
+    if call.arguments.len() < 2 {
+        return;
+    }
+
+    if has_unknown_keywords_or_starred_expr(
+        &call.arguments,
+        &["path", "mode", "dir_fd", "follow_symlinks"],
+    ) {
+        return;
+    }
+
+    let (Some(path_arg), Some(_)) = (path_arg, call.arguments.find_argument_value("mode", 1))
+    else {
+        return;
+    };
+
+    diagnostic.try_set_fix(|| {
+        let (import_edit, binding) = checker.importer().get_or_import_symbol(
+            &ImportRequest::import("pathlib", "Path"),
+            call.start(),
+            checker.semantic(),
+        )?;
+
+        let locator = checker.locator();
+        let path_code = locator.slice(path_arg.range());
+
+        let args = |arg: ArgOrKeyword| match arg {
+            ArgOrKeyword::Arg(expr) => {
+                if expr.range() == path_arg.range() {
+                    None
+                } else {
+                    Some(locator.slice(expr.range()).to_string())
+                }
+            }
+            ArgOrKeyword::Keyword(kw) => match kw.arg.as_deref() {
+                Some("mode") => Some(format!("mode={}", locator.slice(&kw.value))),
+                Some("follow_symlinks") => {
+                    Some(format!("follow_symlinks={}", locator.slice(&kw.value)))
+                }
```

Comment:
Could we use the trick like in the other PR where we slice out the whole `kw`?


```suggestion
                Some("mode" | "follow_symlinks") => locator.slice(kw),
```

then we can avoid the `to_string` call for positional args too.

Reply:
Oh oops, I forgot to wrap it in `Some`, sorry! I think we can still remove `to_string` as long as we remove it from the `ArgOrKeyword::Arg` arm too.